job/job.go

```diff
@@ -141,6 +175,45 @@ func (j *Job) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+// CallbackInfo validates the state of a job and returns a callback info
+// along with an error if appropriate
+func (j *Job) CallbackInfo(downloadURL url.URL) (Callback, error) {
+	var dwURL string
+
+	if j.DownloadState != StateSuccess && j.DownloadState != StateFailed {
+		return Callback{}, fmt.Errorf("Invalid job download state: '%s'", j.DownloadState)
+	}
+
+	if j.DownloadState != StateSuccess {
+		dwURL = ""
+	} else {
+		downloadURL.Path = path.Join(downloadURL.Path, j.Path())
+		dwURL = downloadURL.String()
+	}
+
+	return Callback{
+		Success:      j.DownloadState == StateSuccess,
+		Error:        j.DownloadMeta,
+		Extra:        j.Extra,
+		ResourceURL:  j.URL,
+		DownloadURL:  dwURL,
+		JobID:        j.ID,
+		ResponseCode: j.ResponseCode,
+		Delivered:    true,
+	}, nil
+}
+
+// GetCallbackTypeAndDestination returns callback type and destination from either
+// the job's callback_url or callback_type and callback_dst.
+// When callback_url is present then as callback type the "http" is returned.
+func (j *Job) GetCallbackTypeAndDestination() (string, string) {
```

Comment:
Since this method is only implemented for Notifier, it should belong in the notifier package and not be part of the public API.